src/main/scala/cromwell/engine/backend/sge/SgeBackend.scala

```diff
@@ -0,0 +1,126 @@
+
+        val (qsubStdoutFile, qsubStdoutWriter) = backendCall.callRootPath.resolve("qsub.stdout").fileAndWriter
+        val (qsubStderrFile, qsubStderrWriter) = backendCall.callRootPath.resolve("qsub.stderr").fileAndWriter
+        val process = argv.run(ProcessLogger(qsubStdoutWriter writeWithNewline, qsubStderrWriter writeWithNewline))
+        val rc: Int = process.exitValue()
+        Vector(qsubStdoutWriter, qsubStderrWriter) foreach { _.flushAndClose() }
+        logger.info(s"$tag qsub rc=$rc")
+
+        "^Your job (\\d+)".r.findFirstMatchIn(qsubStdoutFile.slurp) match {
+          case Some(m) =>
+            val sgeJobId = m.group(1).toInt
+
+            def killSgeJob(): Unit = {
+              val (qdelStdoutFile, qdelStdoutWriter) = backendCall.callRootPath.resolve("qdel.stdout").fileAndWriter
+              val (qdelStderrFile, qdelStderrWriter) = backendCall.callRootPath.resolve("qdel.stderr").fileAndWriter
+              val argv = Seq("qdel", sgeJobId.toString)
+              val process = argv.run(ProcessLogger(qdelStdoutWriter writeWithNewline, qdelStderrWriter writeWithNewline))
+              val rc: Int = process.exitValue()
+              Vector(qdelStdoutWriter, qdelStderrWriter) foreach { _.flushAndClose() }
+              logger.debug(s"qsub $sgeJobId (rc=$rc)")
+            }
+
+            backendCall.callAbortRegistrationFunction.register(AbortFunction(killSgeJob))
+
+            @tailrec
+            def waitUntilComplete(): Int = Files.exists(backendCall.rc) match {
+              case true => backendCall.rc.toFile.slurp.stripLineEnd.toInt
+              case false =>
+                logger.info(s"$tag File ${backendCall.rc} does not exist yet")
+                Thread.sleep(5000)
+                waitUntilComplete()
+            }
+
+            val jobRc = waitUntilComplete()
+
+            logger.info(s"$tag File ${backendCall.rc} now exists")
+
+            val stderrFileLength = backendCall.stderr.toFile.length
+
+            if (backendCall.call.failOnStderr && stderrFileLength > 0) {
+              logger.info(s"$tag stderr fail")
+              Failure(new Throwable(s"Workflow ${backendCall.workflowDescriptor.id}: stderr has length $stderrFileLength for command: $instantiatedCommand"))
+            } else {
+              logger.info(s"$tag jobRc=$jobRc")
+              jobRc match {
+                case 0 =>
+                  logger.info(s"$tag postProcess")
+                  postProcess(backendCall)
+                case 143 => Failure(new TaskAbortedException()) // Special case to check for SIGTERM exit code - implying abort
+                case _ => Failure(new Throwable(s"Workflow ${backendCall.workflowDescriptor.id}: return code $rc for command: $instantiatedCommand\n\nFull command was: ${argv.mkString(" ")}"))
+              }
+            }
+
+          case None => Failure(new Throwable("Failed to find SGE job ID in qsub.stdout"))
```

Comment:
Indeed, it's in the CWD of the job, I could just append it to the message though.